charts/prometheus-node-exporter/templates/daemonset.yaml

```
@@ -39,6 +39,8 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       serviceAccountName: {{ include "prometheus-node-exporter.serviceAccountName" . }}
+      nodeSelector:
+        kubernetes.io/os: {{ .Values.nodeselector.os }}
```

I would let the users picks their own generic purpose spec here:

```suggestion
      {{- with .Values.nodeselector }}
      nodeSelector:
        {{- toYaml . | nindent 8 }}
      {{- end }}
```